test/test-TippyTopProvider.js

```diff
@@ -22,19 +23,38 @@ exports["test TippyTopProvider init"] = function(assert) {
 exports["test TippyTopProvider processSite"] = function(assert) {
   // Test init with options
   let tippyTopProvider = new TippyTopProvider({
-    sites: [{
-      url: "https://mozilla.org",
-      image_url: "mozilla-org.png",
-      background_color: "#fff"
-    }, {
-      url: "http://github.com",
-      image_url: "github-com.png",
-      background_color: "#eee"
-    }, {
-      url: "https://www.example.com",
-      image_url: "example-com.png",
-      background_color: "#ddd"
-    }]
+    getSiteData: url => {
+      let sites = {
+        "mozilla.org": {
+          url: "https://mozilla.org",
+          image_url: "mozilla-org.png",
+          background_color: "#fff"
+        },
+        "github.com": {
+          url: "http://github.com",
+          image_url: "github-com.png",
+          background_color: "#eee"
+        },
+        "example.com": {
+          url: "https://www.example.com",
+          image_url: "example-com.png",
+          background_color: "#ddd"
+        }
+      };
+      function getDomain(url) {
+        let domain = urlParse(url, false).host;
+        if (domain && domain.startsWith("www.")) {
+          domain = domain.slice(4);
+        }
+        return domain;
+      }
```

Comment:
Actually, to answer my own ignorant question, the answer seems to be `getDomain()` only returns a normalized `host`, whereas `prettyUrl()` will still return some path info:

```js
const urlParse = require("url-parse");

function getDomain(url) {
  let domain = urlParse(url, false).host;
  if (domain && domain.startsWith('www.')) {
    domain = domain.slice(4);
  }
  return domain;
}

function prettyUrl(url) {
  if (!url) {
    return '';
  }
  return url.replace(/^((https?:)?\/\/)?(www\.)?/i, "").toLowerCase();
}

function diff(url) {
  console.log(`getDomain('${url}'): ${getDomain(url)}`);
  console.log(`prettyUrl('${url}'): ${prettyUrl(url)}`);
}

diff('http://bing.com/foo/bar');
// "getDomain('http://bing.com/foo/bar'): bing.com"
// "prettyUrl('http://bing.com/foo/bar'): bing.com/foo/bar"

diff('https://www.cnn.com');
// "getDomain('https://www.cnn.com'): cnn.com"
// "prettyUrl('https://www.cnn.com'): cnn.com"
```

... and things can get a bit suspect w/ `prettyUrl()` when you have username:password in the URL:

```js
diff('https://user:pass@www.gmail.com/1/somepath');
// "getDomain('https://user:pass@www.gmail.com/1/somepath'): gmail.com"
// "prettyUrl('https://user:pass@www.gmail.com/1/somepath'): user:pass@www.gmail.com/1/somepath"
```